nominal/client.py

```diff
@@ -0,0 +1,481 @@
+                )
+            ),
+            properties={},
+        ),
+    )
+
+
+from _api.timeseries_logicalseries_api import BatchCreateLogicalSeriesRequest, CreateLogicalSeries
+from _api.timeseries_seriescache_api import CachedSeries, CreateCachedSeriesRequest
+
+
+def _create_logical_series(requests: list[CreateLogicalSeries]):
+    return logical_series.batch_create_logical_series(TOKEN, BatchCreateLogicalSeriesRequest(requests=requests))
+
+
+def _create_run(name: str, datasets: list["Dataset"], start_time: int, end_time: int):
+    from _api.scout_run_api import CreateRunRequest, UtcTimestamp, DataSourceRefName, CreateRunDataSource, DataSource
+
+    data_sources = [
+        (ds.filename, CreateRunDataSource(data_source=DataSource(dataset=ds.dataset_rid), series_tags={}))
+        for ds in datasets
+    ]
+    return runs.create_run(
+        TOKEN,
+        CreateRunRequest(
+            title=name,
+            start_time=UtcTimestamp(seconds_since_epoch=start_time),
+            end_time=UtcTimestamp(seconds_since_epoch=end_time),
+            data_sources=dict(data_sources),
+            attachments=[],
+            description="",
+            properties={},
+            labels=[],
+            links=[],
+        ),
+    )
+
+
+def _complete_ingest(dataset_rid: str, start_epoch: int, end_epoch: int, logical_series):
+    from _api.scout_catalog import UpdateBoundsRequest, Bounds, UpdateIngestStatusV2, IngestStatusV2, SuccessResult
+    from _api.datasource import Timestamp as ds_Timestamp, TimestampType as ds_TimestampType
+    from _api.timeseries_seriescache_api import Timestamp as cache_Timestamp
+
+    print("updating bounds")
+    catalog.update_bounds(
+        TOKEN,
+        UpdateBoundsRequest(
+            bounds=Bounds(
+                start=ds_Timestamp(seconds=start_epoch, nanos=0),
+                end=ds_Timestamp(seconds=end_epoch, nanos=0),
+                type=ds_TimestampType.ABSOLUTE,
+            )
+        ),
+        dataset_rid,
+    )
+
+    from _api.timeseries_seriescache_api import CreateCachedSeriesRequest, CachedSeries
+
+    cache_reqs = []
+    for ser in logical_series.responses:
+        cache_reqs.append(
+            CachedSeries(
+                logical_series_rid=ser.rid,
+                start_timestamp=cache_Timestamp(seconds=start_epoch, nanos=0),
+                end_timestamp=cache_Timestamp(seconds=end_epoch, nanos=0),
+                series_data_type=ser.series_data_type,
+            )
+        )
+    print("creating cached series")
+    series_cache.create_cached_series(TOKEN, CreateCachedSeriesRequest(series_to_cache=cache_reqs))
+    print("updating ingest status")
+    catalog.update_dataset_ingest_status_v2(
+        TOKEN,
+        UpdateIngestStatusV2(status=IngestStatusV2(success=SuccessResult()), dataset_uuid=dataset_rid.split(".")[-1]),
+    )
+
+
+class Dataset:
+    def __init__(self, csv_path: str):
```

Comment:
lets create a bunch of classmethods for alternative ctors, e.g. `from_csv()` so that we can later have `from_pandas()`, `from_parquet()`, etc